.github/workflows/pull_request.yml

```diff
@@ -94,3 +94,49 @@ jobs:
 
       - name: Run universe tests
         run: ./mvnw -pl :universe verify -Puniverse -Dmaven.javadoc.skip=true
+
+  compatibility_test:
+    # https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on
+    runs-on: ubuntu-18.04
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Cache local Maven repository
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - uses: satackey/action-docker-layer-caching@v0.0.8
+        continue-on-error: true
+
+      - name: Prepare top of trunk docker image
+        run: ./mvnw clean install -DskipTests -Pcompatibility -Dmaven.javadoc.skip=true -T 1C
+
+      - name: Pull docker image built from master parent commit
+        run: docker pull corfudb/corfu-server:0.3.1-SNAPSHOT
+
+      - uses: actions/checkout@v2
+        with:
+          repository: 'CorfuDB/corfudb-cloud'
+
+      - name: Build universe-core
+        working-directory: ./universe
+        run: rm -r ~/.m2/repository/org/corfudb && ./gradlew clean publishToMavenLocal
```

Comment:
in this case you delete all the artifacts you built, in a previous step

Reply:
it is supposed to do so, as the previous install step will build latest artifacts to maven local, we want the corfudb-cloud to get them from JFrog

Reply:
yeah, but you remove everything in corfu dir when you need only remove universe-core and universe-infrastructure

Reply:
I think the `universe-core` and `universe-infrastructure` are built here. The command is trying to remove runtime, infrastructure, etc., which are supposed to get from jfrog repo

Reply:
oh ok, got it